src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -576,28 +572,28 @@ private List<StopLocation> checkNewStopTimeUpdatesAndFindStops(
       }
 
       // Check arrival time
-      if (stopTimeUpdate.hasArrival() && stopTimeUpdate.getArrival().hasTime()) {
+      final var arrival = addedStopTime.getArrivalTime();
+      if (arrival != null) {
         // Check for increasing time
-        final Long time = stopTimeUpdate.getArrival().getTime();
-        if (previousTime != null && previousTime > time) {
+        if (previousTime != null && previousTime > arrival) {
           debug(tripId, "Trip update contains decreasing times, skipping.");
           return null;
         }
-        previousTime = time;
+        previousTime = arrival;
       } else {
         debug(tripId, "Trip update misses arrival time, skipping.");
         return null;
       }
 
       // Check departure time
-      if (stopTimeUpdate.hasDeparture() && stopTimeUpdate.getDeparture().hasTime()) {
+      final var departure = addedStopTime.getDepartureTime();
+      if (departure != null) {
```

Comment:
Same here.

Actually, convert `getDepartureTime` to return an OptionalInt/OptionalLong and then you can do `departureTime().isPresent()`.